src/components/MoveAdder.ts

```diff
@@ -15,4 +15,247 @@ export class MoveAdder<TGameStartr extends FullScreenPokemon> extends GeneralCom
+                this.gameStarter.menuGrapher.deleteActiveMenu();
+            },
+        );
+    }
+
+    /**
+     * Provides the dialog in the case that a Pokemon must swap out an old move for a new one.
+     * This assumes that the Pokemon already has 4 moves.
+     *
+     * @param pokemon   The pokemon whose moveset is being modified.
+     * @param move   The move that's going to be added into the moveset.
+     */
+    private resolveMoveConflict(pokemon: IPokemon, move: IMove) {
+        this.gameStarter.menuGrapher.addMenuDialog(
+            "GeneralText",
+            [
+                [
+                    pokemon.title.join("") + " is trying to learn " + move.title.toUpperCase() + "!",
+                ],
+                "But, " + pokemon.title.join("") + " can't learn more than 4 moves!",
+                [
+                    "Delete an older move to make room for " + move.title.toUpperCase() + "?",
+                ],
+            ],
+            (): void => {
+                this.gameStarter.menuGrapher.createMenu("Yes/No", {
+                    killOnB: ["GeneralText"],
+                });
+                this.gameStarter.menuGrapher.addMenuList("Yes/No", {
+                    options: [
+                        {
+                            text: "YES",
+                            callback: () => this.acceptLearnMove(pokemon, move),
+                        },
+                        {
+                            text: "NO",
+                            callback: () => this.refuseLearnMove(pokemon, move),
+                        }],
+                });
+                this.gameStarter.menuGrapher.setActiveMenu("Yes/No");
+        });
+    }
+
+    /**
+     * Brings up the dialog for selecting a move to replace with a new move.
+     *
+     * @param pokemon   The pokemon whose moveset is being modified.
+     * @param move   The move that's going to be added into the moveset.
+     */
+    private acceptLearnMove(pokemon: IPokemon, move: IMove) {
+        this.gameStarter.menuGrapher.deleteMenu("Yes/No");
+        this.gameStarter.menuGrapher.createMenu("GeneralText");
+        this.gameStarter.menuGrapher.addMenuDialog(
+            "GeneralText",
+            [
+                [
+                    "Which move should be forgotten?",
+                ],
+            ],
+            (): void => {
+                const moves: IMove[] = pokemon.moves;
+
+                const newPos: IMenuSchemaPosition = {
+                    offset: {
+                        top: -80,
+                        bottom: 80,
+                    },
+                };
+                this.gameStarter.menuGrapher.createMenu("BattleFightList", {
+                    position: newPos,
+                    killOnB: ["GeneralText"],
+                });
+                this.gameStarter.menuGrapher.addMenuList("BattleFightList", {
+                    options: [
```

Comment:
Each of these four options is basically the same except for the index.
Use `moves.map` to only have to write it once.